src/test_l2_hdmi_cec_driver.c

```diff
@@ -1,77 +1,332 @@
 *
-* **Test Group ID:** TODO: Add the group this test belongs to - Basic (for L1): 01 / Module (L2): 02 / Stress (L2): 03)@n
-* **Test Case ID:** TODO: Add the ID of the test case so that it can be logically tracked in the logs@n
+* **Test Group ID:** 02@n
+* **Test Case ID:** 003@n
 *
 * **Test Procedure:**
-* Refer to UT specification documentation [l2_module_test_specification.md](l2_module_test_specification.md)
+* Refer to UT specification documentation [l2_hdmi_cec_driver_test_specification.md](l2_hdmi_cec_driver_test_specification.md)
 */
-void test_l2_hdmi_cec_driver (void)
+
+void test_l2_hdmi_cec_driver_checkLogicalAddress(void)
 {
-	UT_FAIL(This function needs to be implemented!); 
+    UT_LOG("Entering test_l2_hdmi_cec_driver_checkLogicalAddress...");
+
+    int handle;
+    int logicalAddress = 0x0;
+    HDMI_CEC_STATUS status;
+
+    // Open HDMI CEC HAL
+    status = HdmiCecOpen(&handle);
+    UT_LOG(" HdmiCecOpen returned %d", status);
+    UT_ASSERT_EQUAL(status, HDMI_CEC_IO_SUCCESS);
+
+    // Add Logical Address
+    status = HdmiCecAddLogicalAddress(handle, logicalAddress);
+    UT_LOG("Invoking HdmiCecAddLogicalAddress with handle=%d, logicalAddress=%d. Status=%d", handle, logicalAddress, status);    
+    if (status != HDMI_CEC_IO_SUCCESS) 
+    {
+        UT_LOG("Test case failed at HdmiCecAddLogicalAddress");
+        UT_FAIL("Test case failed at HdmiCecAddLogicalAddress");
+    }
+
+    // Remove Logical Address
+    status = HdmiCecRemoveLogicalAddress(handle, logicalAddress);
+    UT_LOG("Invoking HdmiCecRemoveLogicalAddress with handle=%d, logicalAddress=%d. Status=%d", handle, logicalAddress, status);
+    if (status != HDMI_CEC_IO_SUCCESS) 
+    {
+        UT_LOG("Test case failed at HdmiCecRemoveLogicalAddress");
+        UT_FAIL("Test case failed at HdmiCecRemoveLogicalAddress");
+    }
+
+    // Get Logical Address
+    status = HdmiCecGetLogicalAddress(handle, &logicalAddress);
+    UT_LOG("Invoking HdmiCecGetLogicalAddress with handle=%d. Status=%d, logicalAddress=%d", handle, status, logicalAddress);
+    if (status != HDMI_CEC_IO_SUCCESS || logicalAddress != 0x0F) 
+    {
+        UT_LOG("Test case failed at HdmiCecGetLogicalAddress");
+        UT_FAIL("Test case failed at HdmiCecGetLogicalAddress");
+    }
+
+    // Close HDMI CEC HAL
+    status = HdmiCecClose(handle);
+    UT_LOG(" HdmiCecClose returned %d", status);
+    UT_ASSERT_EQUAL(status, HDMI_CEC_IO_SUCCESS);
+
+    UT_LOG("Exiting test_l2_hdmi_cec_driver_checkLogicalAddress...");
+}
+
+/**
+* @brief This test validates the HAL transmission after address removal in HDMI CEC driver
+*
+* This test case is designed to validate the correct functionality of the HDMI CEC driver when a logical address is removed. The test ensures that the driver can still transmit correctly after an address removal operation. This is critical to ensure the robustness of the driver in dynamic HDMI CEC network where devices can be added or removed.
```

Comment:
This statement says that the transmission should be successful even after the removal of the logical address and then sending. But the code below shows that the function HdmiCecTx should fail (not success) in order to pass the test case. Contradictory statements.

Reply:
addressed via updated test requirement